From feebb44edbe53c370d37eb95dac29f973cbc021b Mon Sep 17 00:00:00 2001
From: Louis Chang <louischang@google.com>
Date: Tue, 2 Aug 2022 03:33:39 +0000
Subject: [PATCH] [RESTRICT AUTOMERGE] Do not send new Intent to non-exported
 activity when navigateUpTo

The new Intent was delivered to a non-exported activity while
#navigateUpTo was called from an Activity of a different uid.

Bug: 238605611
Test: atest StartActivityTests
Change-Id: I854dd825bfd9a2c08851980d480d1f3a177af6cf
Merged-In: I854dd825bfd9a2c08851980d480d1f3a177af6cf
(cherry picked from commit 7bd9c8bfb10cb5c37c661b6315ca79ff406bd85b)
(cherry picked from commit ebf6889f753d40d51301a7cd11d6286d4b1a16aa)
Merged-In: I854dd825bfd9a2c08851980d480d1f3a177af6cf
---
 services/core/java/com/android/server/wm/Task.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/wm/Task.java b/services/core/java/com/android/server/wm/Task.java
index d96c04fcc267..67a40ac9dfa4 100644
--- a/services/core/java/com/android/server/wm/Task.java
+++ b/services/core/java/com/android/server/wm/Task.java
@@ -5507,7 +5507,7 @@ class Task extends TaskFragment {
                     abort = true;
                 }
                 if (abort) {
-                    android.util.EventLog.writeEvent(0x534e4554, "238605611", callingUid, "");
+                    Slog.e(TAG, "Cannot navigateUpTo, intent =" + destIntent);
                     foundParentInTask = false;
                 } else {
                     parent.deliverNewIntentLocked(callingUid, destIntent, destGrants,
-- 
2.39.1.456.gfc5497dd1b-goog

